src/Engine/GraphLayout/GraphLayout.cs

```diff
@@ -19,8 +19,28 @@ public class Graph
         public HashSet<Node> Nodes = new HashSet<Node>();
         public HashSet<Edge> Edges = new HashSet<Edge>();
 
+        /// <summary>
+        /// Layers 1 and onwards list the nodes in this graph ordered by layer,
+        /// with smaller numbers to the right part of the graph.
+        /// Layer 0 refers to outside nodes connected to the right of the first layer.
+        /// </summary>
         public List<List<Node>> Layers = new List<List<Node>>();
 
+        /// <summary>
+        /// Edges connected to outside nodes to the left of this graph.
+        /// </summary>
+        public HashSet<Edge> AnchorLeftEdges = new HashSet<Edge>();
+
+        /// <summary>
+        /// Edges connected to outside nodes to the right of this graph.
+        /// </summary>
+        public HashSet<Edge> AnchorRightEdges = new HashSet<Edge>();
```

Comment:
Just wondering if these fields need to be public as I see that they're all being used inside this class?